File: 2262-total-appeal-of-a-string/2262-total-appeal-of-a-string.js
/**
 * @param {string} s
 * @return {number}
 */
var appealSum = function(s) {
  const last = new Array(26).fill(-1);
  const n = s.length;
  let ans = 0;
  
  for (let i = 0; i < n; i++) {
    const key = s.charCodeAt(i) - 'a'.charCodeAt();
    ans += (i - last[key]) * (n - i);
    last[key] = i;    
  }

  return ans;
};

File: 2262-total-appeal-of-a-string/2262-total-appeal-of-a-string.py
class Solution:
    def appealSum(self, s: str) -> int:
        n, ans = len(s), 0
        last = defaultdict(lambda: -1)
        for i, char in enumerate(s):
            ans += (n - i) * (i - last[char])
            last[char] = i
        return ans